From b4c584f3ea1cb8109e51fb1755e244ed0d79870c Mon Sep 17 00:00:00 2001
From: Sameer Thalappil <sameert@qca.qualcomm.com>
Date: Wed, 7 Aug 2013 15:46:55 -0700
Subject: [PATCH] wlan: Dump WCNSS registers upon SSR timer expiry

When SSR timer expires, device is forced to go to download mode. Invoke
WCNSS register dump to collect WCNSS register information that would
give us insight into possible WCNSS failure. Increase the SSR timer,
during SSR notification clients can get into various timeouts while
communicating with various subsystems.

Change-Id: Id800a6eb0a4cf593a02a104ccffc856f518db949
CRs-Fixed: 522868
---
 CORE/HDD/src/wlan_hdd_early_suspend.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/CORE/HDD/src/wlan_hdd_early_suspend.c b/CORE/HDD/src/wlan_hdd_early_suspend.c
index c5f26c6..0dd52a0 100644
--- a/CORE/HDD/src/wlan_hdd_early_suspend.c
+++ b/CORE/HDD/src/wlan_hdd_early_suspend.c
@@ -106,7 +106,7 @@
 #include "wlan_hdd_power.h"
 #include "wlan_hdd_packet_filtering.h"
 
-#define HDD_SSR_BRING_UP_TIME 10000
+#define HDD_SSR_BRING_UP_TIME 180000
 
 static eHalStatus g_full_pwr_status;
 static eHalStatus g_standby_status;
@@ -1336,6 +1336,12 @@ static void hdd_ssr_timer_del(void)
 static void hdd_ssr_timer_cb(unsigned long data)
 {
     hddLog(VOS_TRACE_LEVEL_FATAL, "%s: HDD SSR timer expired", __func__);
+
+#ifdef WCN_PRONTO
+    if (wcnss_hardware_type() == WCNSS_PRONTO_HW)
+        wcnss_pronto_log_debug_regs();
+#endif
+
     VOS_BUG(0);
 }
 
-- 
1.8.2

